feature/policy_forwarding/encapsulation/otg_tests/decap_gre_ipv4/decap_gre_ipv4_test.go

```diff
@@ -0,0 +1,836 @@
+		 	Description: "GRE Decapsulation of IPv6 traffic",
+		 	testFunc:    testGreDecapIPv6,
+		},
+		{
+			Name:        "Testcase-GreDecapIPv4MPLSTraffic",
+			Description: "GRE Decapsulation of IPv4-over-MPLS traffic",
+			testFunc:    testGreDecapIPv4MPLS,
+		},
+		{
+		 	Name:        "Testcase-GreDecapIPv6MPLSTraffic",
+		 	Description: "GRE Decapsulation of IPv6-over-MPLS traffic",
+		 	testFunc:    testGreDecapIPv6MPLS,
+		},
+		{
+			Name:        "Testcase-GreDecapMultiLabelMPLSTraffic",
+		 	Description: "GRE Decapsulation of multi-label MPLS traffic",
+		 	testFunc:    testGreDecapMultiLabelMPLS,
+		},
+		{
+			Name:        "Testcase-GrePassthroughTraffic",
+			Description: "GRE Pass-through (Negative)",
+		 	testFunc:    testGrePassthrough,
+		},
+	}
+
+	// Run the test cases.
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Logf("Description: %s", tc.Description)
+			tc.testFunc(t, dut, otgConfig, config)
+		})
+	}
+
+}
+
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
+	p1 := dut.Port(t, "port1")
+	gnmi.Replace(t, dut, d.Interface(p1.Name()).Config(), configInterfaceDUT(p1, dutPort1, dut))
+	p2 := dut.Port(t, "port2")
+	gnmi.Replace(t, dut, d.Interface(p2.Name()).Config(), configInterfaceDUT(p2, dutPort2, dut))
+
+	// Configure Network instance type on DUT
+	t.Log("Configure/update Network Instance")
+	fptest.ConfigureDefaultNetworkInstance(t, dut)
+
+	if deviations.ExplicitInterfaceInDefaultVRF(dut) {
+		fptest.AssignToNetworkInstance(t, dut, p1.Name(), deviations.DefaultNetworkInstance(dut), 0)
+		fptest.AssignToNetworkInstance(t, dut, p2.Name(), deviations.DefaultNetworkInstance(dut), 0)
+	}
+
+	if deviations.ExplicitPortSpeed(dut) {
+		fptest.SetPortSpeed(t, p1)
+		fptest.SetPortSpeed(t, p2)
+	}
+}
+
+// Configures the given DUT interface.
+func configInterfaceDUT(p *ondatra.Port, a *attrs.Attributes, dut *ondatra.DUTDevice) *oc.Interface {
+	i := a.NewOCInterface(p.Name(), dut)
+	s4 := i.GetOrCreateSubinterface(0).GetOrCreateIpv4()
+	if deviations.InterfaceEnabled(dut) && !deviations.IPv4MissingEnabled(dut) {
+		s4.Enabled = ygot.Bool(true)
+	}
+	i.GetOrCreateSubinterface(0).GetOrCreateIpv6()
+
+	return i
+}
+
+// Congigure Static Routes on DUT
+func configStaticRoute(t *testing.T, dut *ondatra.DUTDevice, prefix string, nexthop string) {
+	s := &oc.Root{}
+	ni := s.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	static := ni.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut))
+	nh := static.GetOrCreateStatic(prefix).GetOrCreateNextHop("0")
+	if nexthop == "Null0" {
+		nh.SetNextHop(oc.UnionString("DROP"))
+	} else {
+		nh.NextHop = oc.UnionString(nexthop)
+	}
+
+	gnmi.Update(t, dut, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Config(), ni)
+}
+
+// Configure static MPLS label binding (LBL1) towards ATE Port 2
+func configureStaticMPLSCLI(t *testing.T, dut *ondatra.DUTDevice) {
```

Comment:
Move to `internal/cfgplugins/mpls.go`

Function should generate the OC.  If the OC is unsupported, a deviation should be created to substitute the CLI.

Example for static MPLS LSP:
https://github.com/openconfig/featureprofiles/blob/5bd6746949359f4a3851e735c9773b040909b65a/feature/mpls/otg_tests/static_lsp_test/static_lsp_test.go#L331

(really, that function belongs in `internal/cfgplugins/mpls.go`,  but you don't need to refactor that test.  Just refactor your existing PR here)

Reply:
Moved the deviation to cfgplugins/mpls.go